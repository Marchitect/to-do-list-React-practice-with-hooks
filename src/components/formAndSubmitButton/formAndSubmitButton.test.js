// formAndSubmitButton.test.js
import React from "react"
import { FormAndSubmitButton } from "./FormAndSubmitButton.component"
import axiosMock from 'axios'
import { render, getByText } from "@testing-library/react"

jest.mock('axios')

it("FormAndSubmitButton renders correctly", () => {
    const formAndSubmitBtn = render(
      <FormAndSubmitButton
        addTodo={axiosMock.get()}
      />
    )
  expect(formAndSubmitBtn).toMatchSnapshot()
})

it("When the Submit Button is clicked, onclick function is triggered", () => {
    const { getByText } = render(
      <FormAndSubmitButton
        addTodo={axiosMock.get()}
      />
    )
  expect(getByText("Submit")).toBeTruthy()
})