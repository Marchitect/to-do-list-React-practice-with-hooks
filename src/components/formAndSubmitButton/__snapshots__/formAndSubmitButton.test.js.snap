// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormAndSubmitButton renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <form
          class="sc-AxhCb hLMWUN"
        >
          <label
            class="sc-AxhUy kbqZXj"
            for="todo-item-input"
          >
            Enter your to-do item here:
          </label>
          <div
            class="sc-AxjAm iaLYws"
          >
            <input
              class="sc-AxirZ cvZsoc"
              name="todo-item-input"
              type="text"
              value=""
            />
            <button
              class="sc-AxiKw glrngo"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <form
        class="sc-AxhCb hLMWUN"
      >
        <label
          class="sc-AxhUy kbqZXj"
          for="todo-item-input"
        >
          Enter your to-do item here:
        </label>
        <div
          class="sc-AxjAm iaLYws"
        >
          <input
            class="sc-AxirZ cvZsoc"
            name="todo-item-input"
            type="text"
            value=""
          />
          <button
            class="sc-AxiKw glrngo"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
